### PR TITLE
remove /style_gram.size in style loss; change tv loss to tensorflow official tv loss function

### PR DESCRIPTION
**Pull request by yihongXU**
Hi,
gram matrix is already normalized by size (HxWxC) :
```python
           bs, height, width, filters = map(lambda i:i.value,layer.get_shape())
            size = height * width * filters
            feats = tf.reshape(layer, (bs, height * width, filters))
            feats_T = tf.transpose(feats, perm=[0,2,1])
            grams = tf.matmul(feats_T, feats) / size
```
So I think that it might be wrong to normalize style loss style_gram.size (= num filters**2), since the stlye loss is already normalized by H^2 x W^2 x C^2 after L2 loss (cf. the formulation of style loss in Johnson's paper.)

some problems with self defined tv loss according to [https://en.wikipedia.org/wiki/Total_variation_denoising](url), so changed it to use tensorflow official tv loss function.

### CODE REVIEW

**Comment by yihongXU** [comment]
sorry, but why should multiply by 2 ?

**Comment by lengstrom** [comment]
I would like to replicate the original loss function as closely as possible.

**Comment by yihongXU** [comment]
please see : 
[https://www.tensorflow.org/api_docs/python/tf/image/total_variation](url)

**Comment by themichaelyang** [comment]
wise up

**Comment by lengstrom** [comment]
Will change myself.
